### DISCUSSION TITLE
Modify osu! ruleset Relax mod to be closer in implementation to stable

### DISCUSSION THREAD

**Discussion by tsunyoku** [ppy/osu#27123] [Ideas]
I know that generally speaking lazer mods (intentionally or otherwise) do not need to match functionality with stable. However, I think there's a valid case that I'd like to hear opinions on for Relax.

Relax has difficulty calculation & performance calculation considerations and has for nearly 2 years, but the issue is that all context built up surrounding relax and it's difficulty come from stable. Relax was implemented in lazer back in 2018 with seemingly no discussion and it's leniency is considerably less than stable's which means the difficulty is not really comparable and trying to adjust for that is unreasonable. The mod in general feels worse to play than standard and the current considerations just don't match that; I think many people weren't aware of this difference.

Relax is still unranked and will be for the foreseeable future as infra doesn't allow for new difficulty attributes at the moment, so I believe it would be an opportunity to make changes before it's too late.

The difference in code is minor, it's only one line that needs to change: https://github.com/ppy/osu/blob/b0420f7ed9e945cb69ffd844664df0705ccc4e8f/osu.Game.Rulesets.Osu/Mods/OsuModRelax.cs#L72-L74

The relevant stable logic can be found [here](https://github.com/peppy/osu-stable-reference/blob/46cd3a10af7cc6cc96f4eba92ef1812dc8c3a27e/osu!/GameModes/Play/Player.cs#L3083-L3086).

Essentially, the change would just be on the if statement to modify the condition and also to bump `relax_leniency` from 3 to 12.

I haven't PR'd as it's probably not a good idea without discussion, but I don't see any reason to keep it how it is currently.

**Comment by bdach** [comment]
I'm fine with adjusting the implementation but if you wish to PR that I'd appreciate something like a replay recorded to use to compare stable and lazer's implementations or something else to be able to make sure that they actually match _empirically_. With changes like these 99% of review overhead is making sure lazer implementation actually _does_ match stable. Just lining up numbers has failed several times on several occasions before.

**Reply by tsunyoku** [reply]
Yeah that sounds reasonable, I'd want to display a side-by-side comparison anyways.
